pulp_smash/tests/platform/cli/test_selinux.py

```diff
@@ -0,0 +1,79 @@
+# coding=utf-8
+"""Tests to verify that Pulp and SELinux are interacting properly."""
+import unittest
+from collections import namedtuple
+
+from pulp_smash import cli, config, utils
+
+
+PS_FIELDS = ('label', 'args')
+
+Process = namedtuple('Process', PS_FIELDS)
+
+
+class ProcessLabelsTestCase(unittest.TestCase):
+    """Test that Pulp processes have correct SELinux labels.
+
+    This test case targets `Pulp Smash #444
+    <https://github.com/PulpQE/pulp-smash/issues/444>`_.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Get all of the processes running on the target Pulp system."""
+        cfg = config.get_config()
+        cmd = ['ps', '-A', '-w', '-w', '-o', ','.join(PS_FIELDS)]
+        if not utils.is_root(cfg):
+            cmd.insert(0, 'sudo')
+        cls.procs = [
+            Process(*line.split(maxsplit=1))
+            for line in cli.Client(cfg).run(cmd).stdout.splitlines()
+        ]
+
+    def _do_test(self, label, arg):
+        """Assert that certain processes have a label of ``label``.
+
+        Select all processes that were invoked with an argument string
+        containing ``arg``. Assert that at least one such process exists, and
+        that all such processes have a label of ``label``.
+        """
+        procs = [proc for proc in self.procs if arg in proc.args]
+        self.assertGreater(len(procs), 0, arg)
+        for proc in procs:
+            # Don't use subTest here. Doing so destroys traceback information
+            # and leads to distractingly verbose output.
+            self.assertEqual(proc.label, label, procs)
+
+    def test_httpd(self):
+        """Verify the labels of the ``wsgi:pulp*`` processes.
+
+        They should have a label of ``unconfined_u:system_r:httpd_t:s0``.
+        """
+        self._do_test('unconfined_u:system_r:httpd_t:s0', 'wsgi:pulp')
+
+    def test_resource_manager(self):
+        """Verify the labels of the ``resource_manager`` processes.
+
+        They should have a label of ``unconfined_u:system_r:celery_t:s0``.
+        """
+        self._do_test('unconfined_u:system_r:celery_t:s0', 'resource_manager')
```

Comment:
Ahh missed the one that does not use it. Anyway adding a constant will help maintenance, but it is not a big deal, we can decide to refactor later.